activesupport/lib/active_support/values/time_zone.rb

```diff
@@ -507,10 +516,21 @@ def yesterday
     end
 
     # Adjust the given time to the simultaneous time in the time zone
-    # represented by +self+. Returns a Time.utc() instance -- if you want an
-    # ActiveSupport::TimeWithZone instance, use Time#in_time_zone() instead.
+    # represented by +self+. Returns a local time with the appropriate offset
+    # -- if you want an ActiveSupport::TimeWithZone instance, use
+    # Time#in_time_zone() instead.
+    #
+    # As of tzinfo 2, utc_to_local returns a Time with a non-zero utc_offset.
     def utc_to_local(time)
-      tzinfo.utc_to_local(time)
+      t = tzinfo.utc_to_local(time)
+      case ActiveSupport.tzinfo_compatibility_version
+      when "1"
+        Time.utc(t.year, t.month, t.day, t.hour, t.min, t.sec, t.sec_fraction)
```

Comment:
Is there a better way to construct this `Time`? I tried a lot of random other stuff 😅